src/ol/control/ScaleLine.js

```diff
@@ -262,8 +262,10 @@ class ScaleLine extends Control {
       Math.log(this.minWidth_ * pointResolution) / Math.log(10));
     let count, width;
     while (true) {
-      count = LEADING_DIGITS[((i % 3) + 3) % 3] *
-          Math.pow(10, Math.floor(i / 3));
+      const decimalCount = Math.floor(i / 3);
+      const decimal = Number(Math.pow(10, decimalCount)
+        .toFixed(Math.abs(decimalCount)));
+      count = LEADING_DIGITS[((i % 3) + 3) % 3] * decimal;
```

Comment:
It's better to apply `toFixed()` to the number when it gets finally rendered, instead of converting it to string and back to number, and then continue doing calculations with it.

Reply:
As the error/inaccuracy occurs right in `Math.pow`, wouldn't it be better to correct it right afterwards before continuing to calculate with the wrong value?

Reply:
No, because we might be running into other floating point issues further down the line. Since we calculate this only for displaying the scale, it's best to do the formatting when displaying the scale.

Reply:
Ok.

Reply:
Resolve due to force push.